### ISSUE TITLE
Error executing guided Topic Model

### ISSUE THREAD

**Issue by AashishTiwari** [MaartenGr/BERTopic#365]
Thanks for all the great work with BERTopic, I was trying the Guided Topic Model from the [documentation](https://maartengr.github.io/BERTopic/tutorial/guided/guided.html) and running into Type Errors which I believe is with NumPy package, similar issue is referenced [here](https://githubmemory.com/repo/MaartenGr/BERTopic/issues/202).

Error Snippet:
```TypeError                                 Traceback (most recent call last)
/tmp/ipykernel_308/1777396079.py in <module>
      1 topic_model = BERTopic(seed_topic_list=wordlist)
----> 2 topics, probs = topic_model.fit_transform(raw_data[:100])

~/.conda/envs/topics/lib/python3.9/site-packages/bertopic/_bertopic.py in fit_transform(self, documents, embeddings, y)
    291         # Reduce dimensionality with UMAP
    292         if self.seed_topic_list is not None and self.embedding_model is not None:
--> 293             y, embeddings = self._guided_topic_modeling(embeddings)
    294         umap_embeddings = self._reduce_dimensionality(embeddings, y)
    295 

~/.conda/envs/topics/lib/python3.9/site-packages/bertopic/_bertopic.py in _guided_topic_modeling(self, embeddings)
   1435         for seed_topic in range(len(seed_topic_list)):
   1436             indices = [index for index, topic in enumerate(y) if topic == seed_topic]
-> 1437             embeddings[indices] = np.average([embeddings[indices], seed_topic_embeddings[seed_topic]], weights=[3, 1])
   1438         return y, embeddings
   1439 

<__array_function__ internals> in average(*args, **kwargs)

~/.conda/envs/topics/lib/python3.9/site-packages/numpy/lib/function_base.py in average(a, axis, weights, returned)
    405             wgt = wgt.swapaxes(-1, axis)
    406 
--> 407         scl = wgt.sum(axis=axis, dtype=result_dtype)
    408         if np.any(scl == 0.0):
    409             raise ZeroDivisionError(

~/.conda/envs/topics/lib/python3.9/site-packages/numpy/core/_methods.py in _sum(a, axis, dtype, out, keepdims, initial, where)
     45 def _sum(a, axis=None, dtype=None, out=None, keepdims=False,
     46          initial=_NoValue, where=True):
---> 47     return umr_sum(a, axis, dtype, out, keepdims, initial, where)
     48 
     49 def _prod(a, axis=None, dtype=None, out=None, keepdims=False,

TypeError: No loop matching the specified signature and casting was found for ufunc add
```

- What have you tried so far?
 -- Reference issue #202 and related guides.
 -- Tried on AWS SageMaker Lab in clean environment from scratch. Also locally on MBP OSX 10.15.7 (Catalina) via Jupyter environments.
 -- Limited Data to just 100 or less samples and seed topic list with only few words from 100 or less samples.

Package Versions:
- Python == 3.9
- BERTopic == 0.9.3
- NumPy == 1.20.3
- Numba == 0.54.1

I have also seen the 0.9.4 version notes and this issue is not referenced / planned for the next release. For the sake of completion, other varieties of Topic Modellers like Supervised (with target categories) and Dynamic TM all are working fine, the issue is only reproduced when running guided TM.

Thanks,
--Aashish

**Comment by MaartenGr** [comment]
Thank you for the extensive description! In all honestly, I completely forgot about this issue since it was closed a while back😅 Did you try setting the dependencies as follows:

```
numba==0.53.1
numpy==1.21.1
```

Reading through the old issue and the one you are currently mentioning I indeed recall it being an issue with Numba >= 0.54 having issues with specific types. I am not entirely sure if restricting the dependencies would actually be the best solution here as there are some specific interactions that have issues. I am afraid that further restriction of dependencies would also have a negative effect on other packages. Also, it seems that only the `np.average` function is the only function affected by this so I think it would be nice if that would be fixed to circumvent any dependency issues. 

The v0.9.4 version will most likely be released somewhere this week and I am not sure if I can fix this before then without extensive experimentation. I think that replacing `np.average` with another function might actually be a quick fix but I will have to do some tests.
